### PR TITLE
refactor(sdk): make settings natively protobuf-friendly

### PR DESCRIPTION
**Pull request by dmitryduev**
Fixes
-----
- Fixes WB-NNNNN
- Fixes #NNNN

Description
-----------
What does the PR do?

### <samp>🤖 Generated by Copilot at c3b80c1</samp>

This pull request simplifies and refactors the console setting for wandb runs, and adds a new Settings message type to the wandb_settings.proto file. The Settings message type contains the configuration options for a wandb run, and can be serialized and deserialized using protobuf. The pull request also updates the wandb_settings module, the wandb_run module, and the wandb_internal_codegen module to use the new Settings message type. Additionally, it fixes a bug in the Azure storage handler that could cause an error when the azure_account_url_to_access_key field is not set in the run settings.

Testing
-------
How was this PR tested?

Checklist
-------
- [ ] Include reference to internal ticket "Fixes WB-NNNNN" and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Ensure PR title compliance with the [conventional commits standards](https://github.com/wandb/wandb/blob/main/CONTRIBUTING.md#conventional-commits)

### <samp>🤖 Generated by Copilot at c3b80c1</samp>

> _We are the wandb warriors, we run with settings of doom_
> _We use the protobuf magic, we generate the code of gloom_
> _We import the right version, we handle the azure keys_
> _We simplify the console, we refactor with ease_

### CODE REVIEW

File: tests/pytest_tests/unit_tests/test_job_builder.py [hunk 1]
```diff
@@ -3,15 +3,28 @@
...
 from wandb.sdk.internal.settings_static import SettingsStatic
+from wandb.sdk.wandb_settings import Settings
 from wandb.util import make_artifact_name_safe
 
 
 def str_of_length(n):
     return "".join(random.choices(string.ascii_uppercase + string.digits, k=n))
 
 
+def make_proto_settings(**kwargs):
```
Comment (kptkin) [inline tests/pytest_tests/unit_tests/test_job_builder.py:18]: todo: remove this use default `to_proto`

File: wandb/sdk/internal/internal_api.py [hunk 2]
```diff
@@ -208,7 +208,8 @@ def __init__(
         }
         self.retry_timedelta = retry_timedelta
         # todo: Old Settings do not follow the SupportsKeysAndGetItem Protocol
-        self.default_settings.update(default_settings or {})  # type: ignore
+        default_settings = default_settings or {}
```
Comment (kptkin) [inline wandb/sdk/internal/internal_api.py:211]: default settings to use a only static settings

File: wandb/sdk/internal/sender.py [hunk 3]
```diff
@@ -1088,8 +1069,10 @@ def _start_run_threads(self, file_dir: Optional[str] = None) -> None:
 
         # hack to merge run_settings and self._settings object together
         # so that fields like entity or project are available to be attached to Sentry events.
-        run_settings = message_to_dict(self._run)
-        self._settings = SettingsStatic({**dict(self._settings), **run_settings})
+        # todo: fixme
```
Comment (kptkin) [inline wandb/sdk/internal/sender.py:1072]: todo fix this

File: wandb/sdk/internal/settings_static.py [hunk 4]
```diff
@@ -80,19 +16,23 @@ def __setitem__(self, key: str, val: object) -> None:
...
-        return str(self.__dict__)
+        return self.proto.__str__()
 
     def __contains__(self, key: str) -> bool:
-        return key in self.__dict__
+        return key in self.keys()
 
-    def get(self, key: str, default: Optional[Any] = None) -> Any:
-        return self.__dict__.get(key, default)
+    def __dict__(self) -> dict:
```
Comment (kptkin) [inline wandb/sdk/internal/settings_static.py:37]: do we need this?

File: wandb/sdk/wandb_init.py [hunk 5]
```diff
@@ -877,12 +877,17 @@ def _attach(
...
         {
             "run_id": attach_id,
-            "_start_time": response["_start_time"],
-            "_start_datetime": response["_start_datetime"],
-            "_offline": response["_offline"],
+            "_start_time": response._start_time.value,
+            "_start_datetime": datetime.fromisoformat(
+                response._start_datetime.value
+            ),  # todo: fixme
+            "_offline": response._offline.value,
```
Comment (kptkin) [inline wandb/sdk/wandb_init.py:890]: fix this!

File: wandb/sdk/wandb_settings.py [hunk 6]
```diff
@@ -1489,6 +1481,48 @@ def make_static(self) -> Dict[str, Any]:
...
+                sequence.value.extend(v)
+            elif isinstance(v, dict):
+                mapping = getattr(settings, k)
+                for key, value in v.items():
+                    # we only support dicts with string values for now
+                    mapping.value[key] = value
+            elif isinstance(v, datetime):
+                time = getattr(settings, k)
+                time.timestamp.FromDatetime(v)
+                time.value = v.isoformat()
```
Comment (kptkin) [inline wandb/sdk/wandb_settings.py:1509]: fix this?

File: wandb/sdk/wandb_settings.py [hunk 7]
```diff
@@ -1489,6 +1481,48 @@ def make_static(self) -> Dict[str, Any]:
...
+                # i.e. None means that the value was not set.
+                pass
+            else:
+                raise TypeError(f"Unsupported type {type(v)} for setting {k}")
+        # TODO: store property sources in the protobuf so that we can reconstruct the
+        #  settings object from the protobuf
+        return settings
+
+    @staticmethod
+    def from_proto(settings: wandb_settings_pb2.Settings) -> "Settings":
```
Comment (kptkin) [inline wandb/sdk/wandb_settings.py:1521]: probably need this for attach

Comment (dmitryduev) [reply wandb/sdk/wandb_settings.py:1459]: 🤷🏻‍♂️ 🤷🏻‍♀️

**Review by kptkin — APPROVED** [review]

**Pull request merged**